addons/sale/models/account_move_line.py

```diff
@@ -20,6 +20,13 @@ def _copy_data_extend_business_fields(self, values):
         super(AccountMoveLine, self)._copy_data_extend_business_fields(values)
         values['sale_line_ids'] = [(6, None, self.sale_line_ids.ids)]
 
+    def _related_analytic_distribution(self):
+        # EXTENDS 'account'
+        vals = super()._related_analytic_distribution()
+        if self.sale_line_ids and not self.analytic_distribution:
+            vals |= self.sale_line_ids.analytic_distribution or {}
```

Comment:
This is raising a singleton error in one of our MMC project, I suppose you could just take the first line if there is multiple?